src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -220,12 +220,12 @@ public async Task<CodeFixCollection> GetDocumentFixAllForIdInSpanAsync(Document
                 return null;
             }
 
-            var result = ArrayBuilder<CodeFixCollection>.GetInstance();
+            using var result = ArrayBuilder<CodeFixCollection>.GetInstance();
             await AppendFixesAsync(document, range, diagnostics, fixAllForInSpan: true, result, cancellationToken).ConfigureAwait(false);
 
             // TODO: Just get the first fix for now until we have a way to config user's preferred fix
             // https://github.com/dotnet/roslyn/issues/27066
-            return result.ToImmutableAndFree().FirstOrDefault();
+            return result.ToImmutable().FirstOrDefault();
```

Comment:
this seems unnecessary expensive :)

Reply:
Yep, didn't look at it but if we believe the TODO will ever (soon) get fixed this is probably the best way of doing things.